snap7/util.py

```diff
@@ -1175,6 +1175,35 @@ def get_date(bytearray_: bytearray, byte_index: int = 0) -> date:
         raise ValueError("date_val is higher than specification allows.")
     return date_val
 
+def set_date(bytearray_: bytearray, byte_index: int, date: date) -> bytearray:
+    """Set value in bytearray to date
+
+        Notes:
+            Datatype `date` consists in the number of days elapsed from 1990-01-01.
+            It is stored as an int (2 bytes) in the PLC.
+
+        Args:
+            bytearray_: buffer to write.
+            byte_index: byte index from where to start writing.
+            date: date object
+
+        Examples:
+            >>> data = bytearray(2)
+
+            >>> snap7.util.set_date(data, 0, date(2024, 3, 27))
+
+            >>> data
+                bytearray(b'\x30\xd8')
+        """
+    if date < date(1990, 1, 1):
+        raise ValueError("date is lower than specification allows.")
+    elif date > date(2168, 12, 31):
+        raise ValueError("date is higher than specification allows.")
+    _days = (date - date(1990, 1, 1)).days
+    _bytes = struct.unpack('2B', struct.pack('>h', _days))
+    bytearray_[byte_index:byte_index + 2] = _bytes
```

Comment:
What if `byte_index` exceeds the length of `bytearray_`?

Reply:
This problem also exists in the other functions I looked (specifically, set_time and set_int, but seems like it's in almost all of them).
Do I need to handle that?

Reply:
I believe not, since this is normal behavior.